engine-nextgen/src/main/kotlin/graphql/nadel/enginekt/util/GraphQLUtil.kt

```diff
@@ -334,9 +349,10 @@ internal suspend fun NadelInstrumentation.beginExecute(
         queryDocument,
         graphQLSchema,
         executionInput.variables,
-        queryDocument.definitions.singleOfType(),
+        queryDocument.getOperationDefinitionOrNull(executionInput.operationName)
```

Comment:
Query document _can_ contain multiple operations. At this point in the execution, we have already validated that the user specified a valid `operationName`, so we just need to pick it out of the document.